test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```diff
@@ -63,23 +63,44 @@ public String getID() {
         try {
             List<Service> serviceList = Exceptions.handleInterruptedCall(
                     () -> dockerClient.listServices(criteria));
-            serviceId = serviceList.get(0).id();
+            log.info("Service list size {}", serviceList.size());
```

Comment:
This change implies `getID()` can return a null value.
Also, looking at the code we can directly use the field `final String serviceName;` as the id, right? This would imply `getID()` directly returns the `serviceName`. 
The logic of fetching the docker specific id for the given `serviceName` using `com.spotify.docker.client.messages.swarm.Service#id` can be a helper function. This would simplify the code further.

Reply:
Fixed